service/api/src/test/java/org/eclipse/kapua/locator/KapuaLocatorExceptionTest.java

```diff
@@ -0,0 +1,155 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eurotech and/or its affiliates and others
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.kapua.locator;
+
+import org.eclipse.kapua.qa.markers.junit.JUnitTests;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(JUnitTests.class)
+public class KapuaLocatorExceptionTest extends Assert {
+
+    String expectedErrorMessage;
+    KapuaLocatorErrorCodes[] kapuaLocatorErrorCodes;
+    Object argument1, argument2, argument3;
+    Throwable[] throwables;
+    String[] expectedMessageNullArguments;
+    String[] expectedMessage;
+
+    @Before
+    public void initialize() {
+        expectedErrorMessage = "kapua-locator-service-error-messages";
+        kapuaLocatorErrorCodes = new KapuaLocatorErrorCodes[]{KapuaLocatorErrorCodes.SERVICE_UNAVAILABLE, KapuaLocatorErrorCodes.SERVICE_PROVIDER_INVALID,
+                KapuaLocatorErrorCodes.FACTORY_UNAVAILABLE, KapuaLocatorErrorCodes.FACTORY_PROVIDER_INVALID, KapuaLocatorErrorCodes.COMPONENT_UNAVAILABLE,
+                KapuaLocatorErrorCodes.COMPONENT_PROVIDER_INVALID, KapuaLocatorErrorCodes.INVALID_CONFIGURATION};
+        argument1 = "argument1";
+        argument2 = "argument2";
+        argument3 = "argument3";
+        throwables = new Throwable[]{new Throwable(), null};
+        expectedMessageNullArguments = new String[]{"Service unavailable {0}", "{0} is not a valid service provider for {1}", "Factory unavailable {0}",
+                "{0} is not a valid factory provider for {1}", "Error: ", "Error: ", "Error: "};
+        expectedMessage = new String[]{"Service unavailable " + argument1, argument1 + " is not a valid service provider for " + argument2, "Factory unavailable " + argument1,
+                argument1 + " is not a valid factory provider for " + argument2, "Error: " + argument1 + ", " + argument2 + ", " + argument3, "Error: " + argument1 + ", " + argument2 + ", " + argument3,
+                "Error: " + argument1 + ", " + argument2 + ", " + argument3};
+    }
+
+    @Test
+    public void kapuaLocatorExceptionKapuaLocatorErrorCodesParameterTest() {
+        for (int i = 0; i < kapuaLocatorErrorCodes.length; i++) {
```

Comment:
Please use enhanced for loop.

Reply:
I used index i to get values from different arrays in the same position